File: Source/ipc/examples/node/writer.js
var IPC = require('../../node/build/Release/IPC');

var conn = new IPC.Connection("ipcdemo", 1);

console.log("Sent: message1");
var msg = new IPC.Message(Buffer.from("Message 1"));
conn.send(msg);

console.log("Sent: message2");
msg = new IPC.Message(Buffer.from("Message 2"));
conn.send(msg);

console.log("Sent: message3");
msg = new IPC.Message(Buffer.from("Message 3"));
conn.send(msg);

File: Source/ipc/examples/java/Writer.java
import com.mittudev.ipc.Connection;
import com.mittudev.ipc.Message;
import com.mittudev.ipc.ConnectionCallback;
import java.lang.Thread;

public class Writer {
	public static void main(String[] args) {
		Connection conn = Connection.connect("ipcdemo", 1);

		System.out.println("Sent: message1");
		Message msg = new Message("Message 1".getBytes());
		conn.send(msg);
		msg.destroy();

		System.out.println("Sent: message2");
		msg = new Message("Message 2".getBytes());
		conn.send(msg);
		msg.destroy();

		System.out.println("Sent: message3");
		msg = new Message("Message 3".getBytes());
		conn.send(msg);
		msg.destroy();

		conn.destroy();
	}
}
